### ISSUE TITLE
Issue with gallery view on collections page

### ISSUE THREAD

**Issue by elizoller** [asulibraries/islandora-repo#286]
the glossary links on the collections page are broken
on keep: https://keep.lib.asu.edu/collections/c

**Comment by elizoller** [comment]
seems to related to complex title block? https://keep.lib.asu.edu/admin/reports/dblog/event/54667

**Comment by wgilling** [comment]
I am pretty sure that this patch to ASUComplexTitle.php would fix the issue here.

```
diff --git a/web/modules/custom/asu_search/src/Plugin/Block/ASUComplexTitle.php b/web/modules/custom/asu_search/src/Plugin/Block/ASUComplexTitle.php
index fd96ed21..2d387b03 100644
--- a/web/modules/custom/asu_search/src/Plugin/Block/ASUComplexTitle.php
+++ b/web/modules/custom/asu_search/src/Plugin/Block/ASUComplexTitle.php
@@ -32,9 +32,6 @@ public function build() {
     $current_route = \Drupal::routeMatch();
     if ($current_route->getParameter('node')) {
       $node = $current_route->getParameter('node');
-      if (!is_object($node)) {
-        $node = \Drupal::entityTypeManager()->getStorage('node')->load($node);
-      }
     }
     elseif ($current_route->getParameter('arg_0')) {
       $node = \Drupal::entityTypeManager()->getStorage('node')->load($current_route->getParameter('arg_0'));
@@ -42,6 +39,10 @@ public function build() {
     else {
       return [];
     }
+    if (!is_object($node)) {
+      $node = \Drupal::entityTypeManager()->getStorage('node')->load($node);
+    }
+
     if ($node->bundle() == "asu_repository_item" || $node->bundle() == "collection") {
       $asu_utils = \Drupal::service('asu_utils');
       $node_is_published = $asu_utils->isNodePublished($node);
```

**Comment by elizoller** [comment]
closed with 08627f70c6e87888d511b8b14e817955872ea0f5
